screens/parkingFlow/Arriving.tsx

```diff
@@ -0,0 +1,201 @@
+import { CompositeScreenProps } from "@react-navigation/native";
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { useCallback, useLayoutEffect, useState } from "react";
+import { View, StyleSheet, SafeAreaView, Platform } from "react-native";
+import MapView, { PROVIDER_GOOGLE } from "react-native-maps";
+import MaterialIcons from "react-native-vector-icons/MaterialIcons";
+
+import { Attribute } from "../booking/BookingSummary";
+import { RegionType } from "../landing/Landing";
+
+import BodyText from "@/components/text/BodyText";
+import LoadingOverlay from "@/components/ui/LoadingOverlay";
+import Colors from "@/constants/color";
+import {
+  AuthenticatedStackParamList,
+  ParkingFlowStackParamList,
+} from "@/types";
+import { formatHumanReadableDateFromDateString } from "@/utils/date";
+
+export type ArrivingProps = CompositeScreenProps<
+  NativeStackScreenProps<ParkingFlowStackParamList, "Arriving">,
+  NativeStackScreenProps<AuthenticatedStackParamList>
+>;
+const Arriving: React.FC<ArrivingProps> = ({ route }) => {
+  const [region, setRegion] = useState<RegionType>();
+  const { bookingRequest, parkingLot } = route.params;
+  useLayoutEffect(() => {
+    getLocation();
+  }, []);
+
+  const getLocation = async () => {
+    const { lat, lng } = parkingLot.coord;
+    setRegion({
+      latitude: lat,
+      longitude: lng,
+      latitudeDelta: 0.0922,
+      longitudeDelta: 0.0421,
+    });
+  };
+
+  const renderAttribute = useCallback(({ attribute, value }: Attribute) => {
+    return (
+      <View style={styles.attributeContainer}>
+        <BodyText
+          text={attribute + ":"}
+          containerStyle={styles.attributeField}
+          textStyle={styles.attributeTextColor}
+        />
+        <BodyText text={value ?? ""} containerStyle={styles.valueField} />
+      </View>
+    );
+  }, []);
+
+  return (
+    <View style={{ flex: 1 }}>
+      {region ? (
+        <>
+          <MapView
+            style={styles.map}
+            initialRegion={region}
+            provider={PROVIDER_GOOGLE}
+          ></MapView>
```

Comment:
Should we show the parking space location ?

Reply:
`const [region, setRegion] = useState<RegionType>();
  const { bookingRequest, parkingLot } = route.params;
  useLayoutEffect(() => {
    getLocation();
  }, []);

  const getLocation = async () => {
    const { lat, lng } = parkingLot.coord;
    setRegion({
      latitude: lat,
      longitude: lng,
      latitudeDelta: 0.0922,
      longitudeDelta: 0.0421,
    });
  };` initialRegion ใส่lat long จากparking lot แล้วมันไม่ได้ใช้แบบนี้ออ

Reply:
I misread it. You did it right